cms/analysis/models.py

```diff
@@ -0,0 +1,226 @@
+                    heading=_("About the data"),
+                ),
+                FieldPanel("contact_details"),
+                FieldPanel("show_cite_this_page"),
+                FieldPanel("main_points"),
+            ],
+            heading=_("Metadata"),
+            icon="cog",
+        ),
+        FieldPanel("headline_figures", icon="data-analysis"),
+        FieldPanel("content", icon="list-ul"),
+    ]
+
+    search_fields: ClassVar[list[index.SearchField | index.AutocompleteField]] = [
+        *BasePage.search_fields,
+        index.SearchField("summary"),
+        index.SearchField("headline_figures"),
+        index.SearchField("content"),
+        index.SearchField("get_admin_display_title", boost=2),
+        index.AutocompleteField("get_admin_display_title"),
+        index.SearchField("news_headline"),
+        index.AutocompleteField("news_headline"),
+    ]
+
+    def clean(self) -> None:
+        """Additional validation on save."""
+        super().clean()
+
+        if self.next_release_date and self.next_release_date <= self.release_date:
+            raise ValidationError({"next_release_date": _("The next release date must be after the release date.")})
+
+    def get_context(self, request: "HttpRequest", *args: Any, **kwargs: Any) -> dict:
+        """Additional context for the template."""
+        context: dict = super().get_context(request, *args, **kwargs)
+        context["table_of_contents"] = self.table_of_contents
+        return context
+
+    def get_admin_display_title(self) -> str:
+        """Changes the admin display title to include the parent title."""
+        return f"{self.get_parent().title}: {self.draft_title or self.title}"
+
+    @property
+    def display_title(self) -> str:
+        """Returns the page display title. If the news headline is set, it takes precedence over the series+title."""
+        return self.news_headline.strip() or self.get_admin_display_title()
+
+    @cached_property
+    def table_of_contents(self) -> list[dict[str, str | object]]:
+        """Table of contents formatted to Design System specs."""
+        items = []
+        for block in self.content:  # pylint: disable=not-an-iterable,useless-suppression
+            if hasattr(block.block, "to_table_of_contents_items"):
+                items += block.block.to_table_of_contents_items(block.value)
+        if self.show_cite_this_page:
+            items += [{"url": "#cite-this-page", "text": _("Cite this analysis")}]
+        if self.contact_details_id:
+            items += [{"url": "#contact-details", "text": _("Contact details")}]
+        return items
+
+    @property
+    def is_latest(self) -> bool:
+        """Returns True if the analysis page is latest in its series based on the release date."""
+        latest_id = (
+            AnalysisPage.objects.sibling_of(self).live().order_by("-release_date").values_list("id", flat=True).first()
+        )
+        return bool(self.pk == latest_id)  # to placate mypy
```

Comment:
This is because we enforce doc strings across the board. I am very much of the opinion they should be required for non-trivial functions, but haven't found a way to conditionally disable them at a pylint config level and I want to keep the inline pylint ignores to a minimum

Reply:
I think that we disable it globally and add it as needed based on our style guide (which we can add). Leave it for now, I'll raise it in tech session with the team.